tests/functional/testConnection.js

```diff
@@ -90,4 +88,41 @@ describe('Connectivity', function() {
     assert(vpn.lastNotification().title === 'VPN Disconnected');
     assert(vpn.lastNotification().message.startsWith('Disconnected from '));
   });
+
+  it.only('Connect to VPN - race condition', async () => {
+    await vpn.setSetting(
+        'serverData',
+        '{"enter_city_name":"","enter_country_code":"","exit_city_name":"Melbourne","exit_country_code":"au"}');
+    await vpn.setSetting('connectionChangeNotification', 'true');
+
+    await vpn.clickOnQuery(queries.screenHome.CONTROLLER_TOGGLE.visible());
```

Comment:
Am I right in thinking this is trying to turn on the VPN?

This could be done at another time, but I think it would be helpful if this was wrapped in a helper function "activate VPN" (and it's sibling deactivate VPN") that assert the current state as on or off then hit click. This would make this test much easier to read.

Reply:
Yes :) we have vpn.activate() but that goes through the inspector. Here I trigger the client on the toggle, which is cooler!

Reply:
I 100% agree that we should be trigger the client with the toggle. I am just saying a helper function e.g.

pseudocode
```
testHelper.toggleOnVpn() {
 toggle = queries.screenHome.CONTROLLER_TOGGLE.visible()
 assert toggle.isOff()
 vpn.clickOnQuery(queries.screenHome.CONTROLLER_TOGGLE.visible());
}
```

So that that way the test code is much easier to read:

```
await vpn.setSetting('connectionChangeNotification', 'true');
await testHelper.toggleOnVpn()
```